ReleaseProcessAutomation.UnitTests/Config/ConfigReaderTests.cs

```diff
@@ -75,4 +76,24 @@ public void GetConfigPathFromBuildProject_InCurrentWorkingDir_GetsProperName ()
 
     Assert.That(configPath, Is.EqualTo("Build/Customizations/ReleaseProcessScript.Test.config"));
   }
+
+  [Test]
+  public void LoadConfig_WithWrongFirstMSBuildPath_LoadsSecondMsBuildPath ()
+  {
+    var reader = new ConfigReader();
+    var configFilePath = Path.Combine(TestContext.CurrentContext.TestDirectory, c_configFileName);
+    var configLines = File.ReadLines(configFilePath);
+    //since the serializer thinks theres only one entry, we can just add another entry of msBuildPath before the typically correct
+    //one to fake an incorrect one, the correct one will be ignored
+    var changedContent = configLines.Select(
+        l => l.Replace(
+            "<!--MsBuildPathTestPlaceholder-->",
+            "<msBuildPath>C:/DEFINETLYNOTANMSBUILDPATH/DEFINETLYNOTANMSBUILD.EXE</msBuildPath>")).ToArray();
+    File.WriteAllLines(configFilePath, changedContent);
+
+    var config = reader.LoadConfig(configFilePath);
+    
+    Assert.That(config.MSBuildSettings.MSBuildPath, Does.Contain("MSBuild.exe"));
+    Assert.That(File.Exists(config.MSBuildSettings.MSBuildPath), Is.True);
```

Comment:
I am still not happy with the way we test this. We rely on a dummy value in the config file to exist and also rely on the fact that the XML is parsed so that the first element is taken. Also if the placeholder is changed or removed the test won't fail.

I propose hardcoding a minimal XML config file that we write to a file to pass the path to the reader. In the config file we have to placeholder values for the file paths and we fill them in before writing the file. We can use `Path.GetTempFile()` (or something like that) to create a file just for the test.

With that we can test the config reading in isolation without the dependency on the actual config.